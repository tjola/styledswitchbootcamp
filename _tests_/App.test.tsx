import React from 'react';

import renderer from 'react-test-renderer';
import App from '@albanian-xrm/styled-switch/App';
import { IAppProps, IStyledSwitchProps } from '@albanian-xrm/styled-switch/App.types';
import { Notifier, SwitchValue } from '@albanian-xrm/styled-switch/notifier';

it('renders correctly', () => {
    const notifier = new Notifier<SwitchValue>();
    const disabledNotifier = new Notifier<boolean>();
    const stylesNotifier = new Notifier<IStyledSwitchProps>();
    const visibleNotifier = new Notifier<boolean>();
    const props: IAppProps = {
        notifier,
        disabledNotifier,
        stylesNotifier,
        visibleNotifier,
        initialVisible: true,
        initialStyles: {
            FalseHandleFill: null,
            FalseHandleImage: null,
            FalseTrackFill: null,
            TrueHandleFill: null,
            TrueHandleImage: null,
            TrueTrackFill: null
        },
        initialValue: true,
        onValueChanged: (value)=>{console.log(value)}
    };

    const tree = renderer
        .create(<App {...props} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
})