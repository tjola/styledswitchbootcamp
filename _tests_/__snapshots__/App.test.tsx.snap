// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="MuiSwitch-root MuiSwitch-sizeMedium css-cm0tsg-MuiSwitch-root"
>
  <span
    className="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <input
      checked={true}
      className="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
    />
  </span>
  <span
    className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
  />
</span>
`;
